---
title: "Untitled"
author: "Gabe Runte"
date: "2/26/2021"
output: html_document
---

```{r setup, include=FALSE}
knitr::opts_chunk$set(echo = TRUE)
library(here)
library(janitor)
library(vegan)
library(phyloseq)
library(metagenomeSeq)
library(lubridate)
library(janitor)
library(calecopal)
library(ape)
library(kmer)
library(bioseq)
library(dada2)
library(matrixStats)
library(tidyverse)
```

```{r, message= FALSE}
size = read_csv(here("data", "ghecto_pilot_sizemetrics.csv"))%>% 
  clean_names()
biomass = read_csv(here("data", "ghecto_pilot_biomass.csv"))%>% 
  clean_names()
community = read_csv(here("data", "seedling_measurements.csv")) %>% 
  clean_names()

pilot.fasta = here("pilot_fasta", "pilot_sequences.fasta")
pilot.fastq = here("pilot_fasta", "pilot_sequences.fastq")

pilot.dnabin = ape::read.dna(pilot.fasta, format= "fasta")

pilot.otus = otu(pilot.dnabin)
pilot.tibble = as_tibble.DNAbin(pilot.dnabin)
pilot.tibble = pilot.tibble %>% 
  mutate(otu = pilot.otus) %>% 
  mutate(for.central = names(pilot.otus)) %>% 
  mutate(is.center = substr(pilot.tibble$for.central, nchar(pilot.tibble$for.central), nchar(pilot.tibble$for.central))) %>% 
  mutate(is.center = is.center == "*")

for(i in 1:length(pilot.tibble[,1])){
  if(substr(pilot.tibble$for.central[i], nchar(pilot.tibble$for.central[i]), nchar(pilot.tibble$for.central[i])) == "*"){pilot.tibble$is.center[i] <- 1} else{pilot.tibble$is.center[i] <- 0}
}
```

```{r}
#unite.seqs = pilot.tibble %>% 
 # filter(is.center == "TRUE") %>% 
#  arrange(otu)

#unite.ref <- "/Volumes/Gabe_Runte_EHD/unite_db/sh_general_release_s_04.02.2020/sh_general_release_dynamic_s_04.02.2020.fasta"
#pilot.seqs <- as.vector(unite.seqs$sequence)
#pilot.taxonomy <- assignTaxonomy(pilot.seqs, unite.ref, multithread = TRUE, tryRC = TRUE)
#write.csv(pilot.taxonomy, here("data", "unite_sedgwick_otus.csv"))
#pilot.taxonomy$sequence <- rownames(pilot.taxonomy)

taxa.tib = read_csv(here("data", "unite_sedgwick_otus.csv")) %>% 
  mutate(otu = seq(1:length(taxa.tib$X1))) %>% 
  rename(otu_seq = X1)

pilot.all = pilot.tibble %>% 
  left_join(taxa.tib) %>%
  separate(label, c("plate", "column", "row")) %>% 
  mutate(row.num = case_when(
    row == "a" ~ "1",
    row == "b" ~ "2",
    row == "c" ~ "3",
    row == "d" ~ "4",
    row == "e" ~ "5",
    row == "f" ~ "6",
    row == "g" ~ "7",
    row == "h" ~ "8"
  )) %>% 
  mutate(column.long = NaN) %>% 
  mutate(column = as.numeric(column)) %>% 
  mutate(row.num = as.numeric(row.num))

for(i in 1:length(pilot.all$plate)){
  if(pilot.all$plate[i] == "1")
  {pilot.all$column.long[i]<- pilot.all$column[i]}
  else{if(pilot.all$plate[i]=="2")
  {pilot.all$column.long[i] <- (pilot.all$column[i]+12)}
    else{pilot.all$column.long[i] <- (pilot.all$column[i]+24)}}
  
}


taxa.mat = matrix(rep(NaN, 8*36), nrow=36, ncol=8)
#taxa.mat[2] = 2

for(i in 1:length(pilot.all$row.num)){
  taxa.mat[pilot.all$column.long[i], pilot.all$row.num[i]] = pilot.all$otu[i]
}


```

```{r setup by root tip}
community= community[1:1438,]
community.subset = community %>% 
  select(seedling_id, seedling_sp, soil_type) %>% 
  rename(id = seedling_id) %>% 
  filter(id %in% size$id)
taxa.mat.tib = tibble(as.data.frame(taxa.mat)) %>% 
  mutate(plate= c(rep(1, 12), rep(2, 12), rep(3, 12))) %>% 
  mutate(row= rep(1:12,3))
size.tax = size %>% 
  separate(plate_position, c("plate", "row")) %>% 
  filter(plate != "na") %>% 
  mutate(plate = as.numeric(plate)) %>% 
  mutate(row = as.integer(row)) %>% 
  left_join(community.subset) %>% 
  left_join(taxa.mat.tib)
```


```{r otu table crafting}
mat.for.otu = as.matrix(size.tax[,9:16])
mat.for.otu[is.nan(mat.for.otu)]<-0

otu.matrix = matrix(rep(NaN, (69*29)), nrow= 29, ncol= 69)
for(i in 1:ncol(otu.matrix)){
 otu.matrix[,i] <- rowCounts(mat.for.otu, value = i)
}
otu.tibble = tibble(as.data.frame(otu.matrix))

pilot.otu.table = size.tax %>% 
  select(1:8) %>% 
  bind_cols(otu.tibble)

otu.nmds = metaMDS(otu.tibble, distance = "bray", k=2) 
otu.fig.tib = tibble(as.data.frame(scores(otu.nmds)))

pilot.otu.table = mutate(pilot.otu.table, nmds1 = otu.fig.tib$NMDS1, nmds2 = otu.fig.tib$NMDS2)


ggplot(pilot.otu.table, aes(nmds1, nmds2,  shape= soil_type, color=seedling_sp)) + 
  geom_point(aes(size= diameter_mm)) +
  geom_jitter()+
  ylim(15,25)+
  xlim(-100,-80)
```

```{r}
taxa.tib.filtered = taxa.tib %>% 
  filter( !is.na(Class))

taxa.tib.reverse.filtered = taxa.tib %>% 
  filter(is.na(Class))

remove.these = which(seq(1,69) %in%  taxa.tib.reverse.filtered$otu)

otu.matrix.filtered = otu.matrix[,-remove.these]
dim(otu.matrix.filtered)

otu.filtered.tib = tibble(as.data.frame(otu.matrix.filtered))
  colnames(otu.filtered.tib)=paste("otu", taxa.tib.filtered$otu, sep = "")
otu.filtered.tib = otu.filtered.tib %>% 
   mutate(otu44 = otu44+otu57,
          otu6 = otu6+otu7+otu8+otu9+otu35+otu36,
          otu14 = otu14+otu15+otu16+otu17+otu18,
          otu41 = otu41+otu62+otu63,
          otu3 = otu3+otu56+otu64,
          otu67 = otu67+otu68) %>% 
   select(-c(otu57, 
             otu7, otu8, otu9, otu35, otu36,
             otu15, otu16, otu17, otu18,
             otu62, otu63,
             otu56, otu64,
             otu68))
otu.filtered.tib = otu.filtered.tib[-c(1,12),]


pilot.otu.table.filtered = size.tax %>% 
  select(1:8) 
pilot.otu.table.filtered = pilot.otu.table.filtered[-c(1,12),]
pilot.otu.table.filtered = pilot.otu.table.filtered %>% 
  bind_cols(otu.filtered.tib)

otu.nmds.filtered = metaMDS(otu.filtered.tib, distance = "bray", k=2) 
otu.fig.tib.filtered = tibble(as.data.frame(scores(otu.nmds.filtered)))

pilot.otu.table.filtered = mutate(pilot.otu.table.filtered, nmds1 = otu.fig.tib.filtered$NMDS1, nmds2 = otu.fig.tib.filtered$NMDS2)


ggplot(pilot.otu.table.filtered, aes(nmds1, nmds2,  shape= soil_type, color=seedling_sp)) + 
  geom_point(aes(size= diameter_mm)) +
  geom_jitter()+
  ylim(-5,-7.5)+
  xlim(-28,-18)
```

```{r}
oaks.otu = pilot.otu.table.filtered %>% 
  filter(seedling_sp == "Q")

colSums(as.matrix(oaks.otu[,9:30]))
rowSums(as.matrix(oaks.otu[,9:30]))


bigcone.otu = pilot.otu.table.filtered %>% 
  filter(seedling_sp == "P")
colSums(as.matrix(bigcone.otu[,9:30]))
rowSums(as.matrix(bigcone.otu[,9:30]))
```

