---
title: "sequence processing"
author: "Gabe Runte"
date: "2024-01-12"
output: html_document
---

```{r setup, include=FALSE}
knitr::opts_chunk$set(echo = TRUE)

library(RColorBrewer)
library(vegan)
library(here)
library(janitor)
library(gridExtra)
library(tidyverse)

```


# First, I wrangle the data from Geneious
```{r}
folder = here("data/sanger_data/total_project")

# all trimmed sequences (bad sequences would have been removes hopefully)
all_seqs = read_csv(here(folder, "all_seqs.csv")) %>% 
  clean_names() %>% select(1:3) %>% 
  filter(str_length(sequence)>100)

# the consensus sequence associated with each contig from the assembly
consensus = read_csv(here(folder, "consensus.csv"))%>% 
  clean_names()%>% select(1:2)%>% 
  rename(contig = name, contig_seq = sequence) %>% 
  filter(str_length(contig_seq)>100) #remove any useless contigs

# a mapping of sequence names onto contig assignments
contigs = read_csv(here(folder, "contigs.csv"))%>% 
  clean_names() %>% select(1,3) %>% 
  rename(contig = sequence_list_name)

# mash them all into 1 sheet with all important information
seq = all_seqs %>% 
  left_join(contigs) %>% 
  left_join(consensus) %>% 
  mutate(unite_seq = case_when(
    is.na(contig)~ sequence, 
    !is.na(contig) ~ contig_seq))
  
```

```{r, eval = F}
morpho_sample_sheet = read_csv(here("data/harvest", "harvest_tipseq_printout.csv")) %>% 
  mutate(sampleset = "m5x") %>% 
  bind_rows(read_csv(here("data/harvest", "harvest_tipseq_printout2.csv")) %>% 
  mutate(sampleset = "m2x")) %>% 
  bind_rows(read_csv(here("data/harvest", "harvest_tipseq_printout3.csv")) %>% 
  mutate(sampleset = "m3x")) %>%  
  bind_rows(read_csv(here("data/harvest", "harvest_tipseq_printout4.csv")) %>% 
  mutate(sampleset = "m4x")%>% 
    rename(comb_column = newcolumn, comb_row = newrow)) %>% 
  rename(morpho_plate = sampleset, morpho_column = comb_column, morpho_row = comb_row) %>% 
  select(plate, column, row, morpho_plate, morpho_column, morpho_row)

#write_csv(morpho_sample_sheet, here("data", "harvest", "morpho_set_mapping.csv"))


repot_sample_sheet_initial = read_csv(here("data/sanger_data", "nov2023_repotting_seqset.csv")) 
repot_sample_sheet_build = read_csv(here("data/sanger_data", "nov2023_repotting_seqset.csv")) %>% 
  left_join(tibble(expand_grid(unique(repot_sample_sheet_initial$column), c("A", "B", "C", "D", "E", "F", "G", "H"))) %>% 
              rename(column= 1, row = 2))%>% 
  mutate(re_row = row) %>% 
  select(2:3, 6:9) %>% 
  mutate(re_plate = paste0("repot", str_sub(re_plate, -1)))
repot5 = read_csv(here("data/sanger_data", "nov2023_repotting_morphoset.csv")) %>% 
  rename(re_plate = n_plate, re_column = n_column, re_row = n_row) %>% 
  mutate(plate = plate +219000)%>% 
  mutate(re_plate = paste0("repot", str_sub(re_plate, -1)))
repot6 = read_csv(here("data/sanger_data", "repot6_set.csv")) %>% 
  select(3:8)

repot_map = bind_rows(repot_sample_sheet_build, repot5, repot6) %>% 
  relocate(row, .before = re_plate)

#write_csv(repot_map, here("data", "repotting", "repot_set_mapping.csv"))
  
```

# Next I need to map all of these sequences back to their relevance to the project by assigning them to plates, wells, and their associated seedling information. 
```{r}
#mapping sets 
repot_map = read_csv(here("data", "repotting", "repot_set_mapping.csv"))
morpho_map = read_csv(here("data", "harvest", "morpho_set_mapping.csv"))

morpho_set = seq %>% 
  filter(str_sub(name, 5,5) == "m") %>% 
  separate(name, into = c("project", "morpho_plate", "morpho_column", "morpho_row")) %>% 
  mutate(morpho_column = as.numeric(morpho_column)) %>% 
  left_join(morpho_map)%>% 
  mutate(timepoint = "harvest")

repot_set = seq %>% 
  filter(str_sub(name, 5,9) == "repot")%>% 
  separate(name, into = c("project", "re_plate", "re_column", "re_row")) %>% 
  mutate(re_column = as.numeric(re_column)) %>% 
  left_join(repot_map)%>% 
  mutate(timepoint = "repot")

plate_set = seq %>% 
  filter(str_sub(name, 1,1) == "p")%>% 
  mutate(plate = 219000 + as.numeric(str_sub(name, 2,3))) %>% 
  mutate(name = case_when(
    str_sub(name, -11,-11)!="-" ~ paste0(str_sub(name,1,-11),"-", str_sub(name,-10)),
    .default = name
  )) %>% 
  mutate(name = case_when(
    str_length(name)==22 ~ name,
    str_length(name)==21 ~ paste0(str_sub(name,1,4),0, str_sub(name, 5))
  )) %>% 
  mutate(column = case_when(
    str_length(name)==21~ str_sub(name, 5,5),
    str_length(name)==22~ str_sub(name, 5,6)
    )) %>% 
  mutate(row = case_when(
    str_length(name)==21~ str_sub(name, 6,6),
    str_length(name)==22~ str_sub(name, 7,7))) %>% 
  mutate(column = as.numeric(column)) %>% 
  mutate(timepoint = "repot")
```

```{r}
treat = read_csv(here("data/design", "seedling_treatments.csv"))

repot_tips = read_csv(here("data", "repotting", "ghecto_repotting_root_pulling_data.csv"))%>%   pivot_longer(A:H, names_to = "row", values_to = "morphotype" )  %>% 
  filter(!is.na(morphotype)) %>% 
  left_join(treat)%>% 
  mutate(morphotype = str_sub(morphotype, 1,1))%>% 
  select(seedling_id, soil_type, seedling_sp, plate, column, row, morphotype)

harvest_tips = read_csv(here("data/harvest", "harvest_roottips - all.csv"))%>%   pivot_longer(A:H, names_to = "row", values_to = "morphotype" )  %>% 
  filter(!is.na(morphotype)) %>% 
  mutate(seedling_id = as.numeric(seedling_id)) %>% 
  left_join(treat)%>% 
  mutate(morphotype = str_sub(morphotype, 1,1))%>% 
  select(seedling_id, soil_type, seedling_sp, plate, column, row, morphotype)


all_tips = bind_rows(repot_tips, harvest_tips)


seq_map = bind_rows(plate_set, repot_set, morpho_set) %>% 
  select(colnames(plate_set), -name) %>% 
  filter(!is.na(plate)) %>% 
  relocate(c(plate, column, row, timepoint), 1) %>% 
  mutate(row = toupper(row)) %>% 
  left_join(all_tips)
```

```{r, eval = FALSE}
unite.ref <- "/Users/Gabe/Desktop/phd/ghecto/unite_db/sh_general_release_s_10.05.2021/sh_general_release_dynamic_s_10.05.2021.fasta"
unique_sequences = unique(seq_map$unite_seq)
tiptaxa = dada2::assignTaxonomy(unique_sequences, unite.ref, multithread = TRUE, tryRC = TRUE)


unite_taxa = tibble(as.data.frame(tiptaxa)) %>% 
  mutate(unite_seq = rownames(tiptaxa))
write_csv(unite_taxa, here(folder, "unite_output.csv"))
```

```{r}
unite_taxa = read_csv(here(folder, "unite_output.csv")) %>% 
  clean_names()
seq_taxa = seq_map %>% 
  left_join(unite_taxa)

seq_clean = seq_taxa %>% 
  filter(!is.na(family)) %>% 
  mutate(taxonomy = paste(kingdom, phylum, class, order, family, genus, species, sep = ";")) %>% 
  mutate(sample = paste(plate, column, row, sep = "_"))




fung_df = seq_clean %>% 
  select(sample, taxonomy)

write_csv(fung_df, here(folder, "taxa_funguild.csv"))
#open file with xcel or other spreadsheet software then convert to .txt file at same file path (see below)
here(folder, "taxa_funguild.txt")
```
 

```{bash, engine.opts='-l', eval = F}

python /Users/Gabe/Desktop/phd/FUNGuild-master/Guilds_v1.1.py -otu /Users/Gabe/Desktop/phd/ghecto/data/sanger_data/total_project/taxa_funguild.txt

```
 
 
```{r}
guilds = read_csv(here(folder, "taxa_funguild.guilds.csv"))

seq_full = seq_clean %>% 
  left_join(guilds)

write_csv(seq_full, here(folder, "full_project_sangerdata.csv"))
```
 
